public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx

```diff
@@ -25,37 +26,41 @@ export const OutsideRangePlugin = ({ config, onChangeTimeRange }: ThresholdContr
     });
   }, [config]);
 
+  // If we don't have enough time values or if we can't
+  // change the time range then we don't display zoom to data
   if (timevalues.length < 2 || !onChangeTimeRange) {
     return null;
   }
 
+  // If we don't have an appropriate time range then we don't know if data is
+  // being displayed
   if (!timeRange || !timeRange.time || !timeRange.min || !timeRange.max!) {
     return null;
   }
 
   // Time values are always sorted for uPlot to work
-  let i = 0,
-    j = timevalues.length - 1;
+  let i = 0;
+  let j = timevalues.length - 1;
 
-  while (i <= j && timevalues[i] == null) {
-    i++;
-  }
+  // Increment i until we get the last non-null value
+  for (; i < timevalues.length && timevalues[i] == null; i++) {}
 
-  while (j >= 0 && timevalues[j] == null) {
-    j--;
-  }
+  // Decrement j until we get the first non-null-value
+  for (; j >= 0 && timevalues[j] == null; j--) {}
 
-  const first = timevalues[i];
-  const last = timevalues[j];
-  const fromX = timeRange.min;
-  const toX = timeRange.max;
+  // Grab the first and last time values
+  const last = timevalues[i];
+  const first = timevalues[j];
 
+  // If we only have null data then there's no data to zoom
   if (first == null || last == null) {
     return null;
   }
 
-  // (StartA <= EndB) and (EndA >= StartB)
-  if (first <= toX && last >= fromX) {
+  // If the first time is less than or equal to the top of time in the range
+  // or the last time is greater than or equal to the bottom of the time range
+  // then there is data to be displayed so we return null
+  if (first <= timeRange.max || last >= timeRange.min) {
```

Comment:
This was the line in question causing the error, the `&&` would cause the message to be displayed in cases where data was outside of one end of the time range but not the other

Reply:
the message then was technically correct, just not wanted? there was actually data beyond the view range? in that case should we just be silent about it completely?

Reply:
We could make the behavior a bit better perhaps by having another condition which ends up showing a message like "Show all data" or something like that. However we'd probably need to update the design to not be displayed in the center of the panel.